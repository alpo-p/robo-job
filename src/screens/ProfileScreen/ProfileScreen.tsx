import React from 'react'
import { Text } from 'react-native'
import { SafeContainer } from '../../sharedComponents/SafeContainer'

const ProfileScreen = () => (
  <SafeContainer>
    <Text>Profile is rendered here</Text>
  </SafeContainer>
)

export default ProfileScreen
