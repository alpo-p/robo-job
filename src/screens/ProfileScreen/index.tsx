import { useTheme } from '@react-navigation/native'
import React, { useContext, useRef, useState } from 'react'
import {
  Alert,
  Keyboard,
  KeyboardAvoidingView,
  Platform,
  ScrollView,
  View,
} from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import { VerticalSpaceOf16 } from '../../common/components/VerticalSpaceOf16'
import { VerticalSpaceOf24 } from '../../common/components/VerticalSpaceOf24'
import { GlobalBooleansContext } from '../../contexts/GlobalBooleansProvider'
import HelloText from './components/HelloText'
import InputComponent from './components/InputComponent'
import SaveProfileButton from './components/SaveProfileButton'

const WHITE_BOX_BASE_SIZE = 48

const PLACEHOLDERS = {
  name: 'Jared Dunn',
  summary: `I am a business graduate with a degree in Marketing looking for work in the field.
I have worked as a marketing generalist in a small business handling all sides, such as updating websites and copywriting.
People say I am diligent and easy to work with. I'm looking for full-time offers.`,
  xp: `- Marketing coordinator: Pied Piper (2017-2021)
- Cashier: K-Market Ilola (2010-2017)`,
  education: `- Msc. Marketing - Trolls of Trinity Christian College in Illinois (2010)
- High school degree - Nowheria in the middle of NowWhere (2005)`,
  links: 'www.linked.com/alpopanula',
}

export default () => {
  const { colors } = useTheme()
  const viewRef = useRef<ScrollView | null>(null)
  const { setGlobalBooleans } = useContext(GlobalBooleansContext)

  const [name, setName] = useState('')
  const [summary, setSummary] = useState('')
  const [experience, setExperience] = useState('')
  const [education, setEducation] = useState('')
  const [links, setLinks] = useState('')

  const onTextInputFocus = () => {
    console.log('textinputfocus on profile')
  }

  const saveProfile = () => {
    console.log('saving profile')
    Keyboard.dismiss()
    setGlobalBooleans(s => {
      const state = { ...s }
      state.showProfileBadge = false
      return state
    })
    Alert.alert('Saved profile!')
  }

  return (
    <SafeAreaView>
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
      >
        <ScrollView
          style={{
            backgroundColor: colors.primary,
            margin: 8,
            paddingTop: 24,
            paddingBottom: 44,
            height: '99%',
            borderRadius: 15,
            borderColor: colors.primary,
            borderWidth: 4,
          }}
          ref={viewRef}
        >
          <HelloText />
          <InputComponent
            title="my name is"
            height={WHITE_BOX_BASE_SIZE}
            placeholder={PLACEHOLDERS.name}
            value={name}
            onChangeText={setName}
            onFocus={onTextInputFocus}
          />
          <VerticalSpaceOf16 />
          <InputComponent
            title="who am I"
            height={WHITE_BOX_BASE_SIZE * 4}
            placeholder={PLACEHOLDERS.summary}
            value={summary}
            onChangeText={setSummary}
            onFocus={onTextInputFocus}
            multiline
          />
          <VerticalSpaceOf16 />
          <InputComponent
            title="linkedin or other links"
            height={WHITE_BOX_BASE_SIZE * 2}
            placeholder={PLACEHOLDERS.links}
            value={links}
            onChangeText={setLinks}
            onFocus={onTextInputFocus}
            multiline
          />
          <VerticalSpaceOf16 />
          <InputComponent
            title="work experience"
            height={WHITE_BOX_BASE_SIZE * 3}
            placeholder={PLACEHOLDERS.xp}
            value={experience}
            onChangeText={setExperience}
            onFocus={onTextInputFocus}
            multiline
            optional
          />
          <VerticalSpaceOf16 />
          <InputComponent
            title="schools"
            height={WHITE_BOX_BASE_SIZE * 3}
            placeholder={PLACEHOLDERS.education}
            value={education}
            onChangeText={setEducation}
            onFocus={onTextInputFocus}
            multiline
            optional
          />
          <VerticalSpaceOf24 />
          <SaveProfileButton onPress={saveProfile} />
          <View
            style={{
              marginTop: 24 * 5,
            }}
          />
        </ScrollView>
      </KeyboardAvoidingView>
    </SafeAreaView>
  )
}
