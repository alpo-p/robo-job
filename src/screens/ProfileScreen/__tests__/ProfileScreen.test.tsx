import React from 'react'
import create from '../../../testUtils/rendererCreateWithNavigation'
import ProfileScreen from '../index'

describe('ProfileScreen', () => {
  it('should match snapshot', () => {
    const tree = create(<ProfileScreen />)
    expect(tree).toMatchSnapshot()
  })
})
