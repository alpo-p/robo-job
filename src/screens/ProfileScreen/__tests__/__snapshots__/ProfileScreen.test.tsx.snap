// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen should match snapshot 1`] = `
<RNCSafeAreaView>
  <View
    onLayout={[Function]}
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <RCTScrollView
      keyboardShouldPersistTaps="handled"
      style={
        Object {
          "backgroundColor": "#000000",
          "borderColor": "#000000",
          "borderRadius": 15,
          "borderWidth": 4,
          "height": "99%",
          "margin": 8,
          "paddingBottom": 44,
          "paddingTop": 24,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignSelf": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "DamascusBold",
                "fontSize": 40,
                "textTransform": "uppercase",
              }
            }
          >
            HELLO
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginLeft": 16,
              "marginRight": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Damascus",
                "fontSize": 20,
                "textTransform": "none",
              }
            }
          >
            my name is
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Jared Dunn"
            placeholderTextColor="#d0d0d0"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 2,
                "color": "rgb(28, 28, 30)",
                "fontSize": 16,
                "height": 48,
                "padding": 12,
                "textAlign": "center",
                "width": "100%",
              }
            }
            textAlign="center"
            textAlignVertical="top"
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "DamascusLight",
                "fontSize": 14,
                "textTransform": "none",
              }
            }
          >
            recommended to fill
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 16,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "marginLeft": 16,
              "marginRight": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Damascus",
                "fontSize": 20,
                "textTransform": "none",
              }
            }
          >
            who am I
          </Text>
          <TextInput
            allowFontScaling={true}
            multiline={true}
            onChangeText={[Function]}
            placeholder="I am a business graduate with a degree in Marketing looking for work in the field.
I have worked as a marketing generalist in a small business handling all sides, such as updating websites and copywriting.
People say I am diligent and easy to work with. I'm looking for full-time offers."
            placeholderTextColor="#d0d0d0"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 2,
                "color": "rgb(28, 28, 30)",
                "fontSize": 16,
                "height": 192,
                "padding": 12,
                "textAlign": "left",
                "width": "100%",
              }
            }
            textAlign="center"
            textAlignVertical="top"
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "DamascusLight",
                "fontSize": 14,
                "textTransform": "none",
              }
            }
          >
            recommended to fill
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 16,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "marginLeft": 16,
              "marginRight": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Damascus",
                "fontSize": 20,
                "textTransform": "none",
              }
            }
          >
            linkedin or other links
          </Text>
          <TextInput
            allowFontScaling={true}
            multiline={true}
            onChangeText={[Function]}
            placeholder="www.linked.com/alpopanula"
            placeholderTextColor="#d0d0d0"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 2,
                "color": "rgb(28, 28, 30)",
                "fontSize": 16,
                "height": 96,
                "padding": 12,
                "textAlign": "left",
                "width": "100%",
              }
            }
            textAlign="center"
            textAlignVertical="top"
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "DamascusLight",
                "fontSize": 14,
                "textTransform": "none",
              }
            }
          >
            recommended to fill
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 16,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "marginLeft": 16,
              "marginRight": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Damascus",
                "fontSize": 20,
                "textTransform": "none",
              }
            }
          >
            work experience
          </Text>
          <TextInput
            allowFontScaling={true}
            multiline={true}
            onChangeText={[Function]}
            placeholder="- Marketing coordinator: Pied Piper (2017-2021)
- Cashier: K-Market Ilola (2010-2017)"
            placeholderTextColor="#d0d0d0"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 2,
                "color": "rgb(28, 28, 30)",
                "fontSize": 16,
                "height": 144,
                "padding": 12,
                "textAlign": "left",
                "width": "100%",
              }
            }
            textAlign="center"
            textAlignVertical="top"
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "DamascusLight",
                "fontSize": 14,
                "textTransform": "none",
              }
            }
          >
            (optional)
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 16,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "marginLeft": 16,
              "marginRight": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Damascus",
                "fontSize": 20,
                "textTransform": "none",
              }
            }
          >
            schools
          </Text>
          <TextInput
            allowFontScaling={true}
            multiline={true}
            onChangeText={[Function]}
            placeholder="- Msc. Marketing - Trolls of Trinity Christian College in Illinois (2010)
- High school degree - Nowheria in the middle of NowWhere (2005)"
            placeholderTextColor="#d0d0d0"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 2,
                "color": "rgb(28, 28, 30)",
                "fontSize": 16,
                "height": 144,
                "padding": 12,
                "textAlign": "left",
                "width": "100%",
              }
            }
            textAlign="center"
            textAlignVertical="top"
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "DamascusLight",
                "fontSize": 14,
                "textTransform": "none",
              }
            }
          >
            (optional)
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 24,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#087c61",
              "borderRadius": 8,
              "opacity": 1,
              "padding": 16,
              "width": "91%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "DamascusBold",
                "fontSize": 16,
                "textTransform": "uppercase",
              }
            }
          >
            Save
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 120,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaView>
`;
